### PR TITLE
Added chucker dev extension to help debug network requests

### PR DESCRIPTION
**Pull request by chege4179**

### CODE REVIEW

**Comment by chege4179** [comment]
@akitikkx

**Comment by akitikkx** [comment]
Hi @chege4179 

Thank you for the PR.

Please explain the description why you think the app needs this - what it will help solve etc

Thanks!

**Comment by chege4179** [comment]
The app relies heavily on making network requests to fetch data and if network requests fail this extension will help find what's went wrong since the extension logs all HTTP requests made by the app through the interceptor and we can inspect why the request failed 
This can help accelerate error detection in development

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -131,6 +131,11 @@ timber = { module = "com.jakewharton.timber:timber", version.ref = "timber" }
 test-harness = { module = "com.google.accompanist:accompanist-testharness", version.ref = "testHarness" }
 work-runtime-ktx = { module = "androidx.work:work-runtime-ktx", version.ref = "workRuntimeKtx" }
 
+chucker-debug =  "com.github.chuckerteam.chucker:library:4.0.0"
```
Comment (akitikkx) [inline gradle/libs.versions.toml:134]: Please check the above definitions as references which follow the format
```
[versions]
...
<library_version> = 4.0.0
...
[libraries]
...
name = {module = "<module name>", version.ref="<library_version>"}
```
And are also defined in alphabetical order

**Comment by akitikkx** [comment]
> The app relies heavily on making network requests to fetch data and if network requests fail this extension will help find what's went wrong since the extension logs all HTTP requests made by the app through the interceptor and we can inspect why the request failed This can help accelerate error detection in development

I haven't used Chucker before so I will need to look into it first

File: gradle/libs.versions.toml (hunk 1 above)
Comment (chege4179) [reply gradle/libs.versions.toml:134]: Ok will make the necessary changes

**Comment by chege4179** [comment]
> 

You learn more about it here 
https://github.com/ChuckerTeam/chucker

File: app/build.gradle [hunk 2]
```diff
@@ -235,6 +235,9 @@ dependencies {
     implementation libs.profileinstaller
     baselineProfile project(path: ':baselineprofile')
 
+    debugImplementation libs.chucker.debug
+    releaseImplementation libs.chucker.release
```
Comment (akitikkx) [inline app/build.gradle:239]: Inspecting the network traffic is fine on a debug version of the app but not on a release version. Let's keep this only for debug

**Review by akitikkx — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (chege4179) [reply app/build.gradle:239]: The release version isolates chucker from the release builds that's why its needed

Comment (akitikkx) [reply app/build.gradle:239]: I don't understand what you mean by "isolates"?

Users of the production release should not see any debuggable information. This is the same approach with leakcanary. The `release` version is not debuggable

Comment (chege4179) [reply app/build.gradle:239]: I mean the debuggable information will not be available for the release builds that's why is included

Comment (akitikkx) [reply app/build.gradle:239]: I checked the "Getting started" section documentation and I see they insist on having both `debug` and `release` defined in the dependencies. I will need to see its behavior on a release build so I am certain the production users are not presented with any debuggable UI

Comment (akitikkx) [reply app/build.gradle:239]: > I mean the debuggable information will not be available for the release builds that's why is included

Got you. Going to need to test this on a release build first

Comment (chege4179) [reply app/build.gradle:239]: No problem

**Comment by chege4179** [comment]
Hey @akitikkx 
Were you able to test it on a release build

**Comment by akitikkx** [comment]
> Hey @akitikkx Were you able to test it on a release build

Hey @chege4179 . Not yet. I have been attending to other things I am afraid.

There are also merge conflicts, please address them?

**Comment by chege4179** [comment]
I have resolved the conflict

**Review by akitikkx — APPROVED** [review]

**Pull request merged**